library/core/src/main/java/com/google/android/exoplayer2/text/ssa/SsaStyle.java

```diff
@@ -114,7 +120,8 @@ public static SsaStyle fromStyleLine(String styleLine, Format format) {
       return new SsaStyle(
           styleValues[format.nameIndex].trim(),
           parseAlignment(styleValues[format.alignmentIndex].trim()),
-          parseColor(styleValues[format.primaryColorIndex].trim()));
+          parseColor(styleValues[format.primaryColorIndex].trim()),
+          parseFontSize(styleValues[format.fontSizeIndex].trim()));
```

Comment:
Is it better to use `Cue.DIMEN_UNSET` here if `fontSizeIndex == C.INDEX_UNSET`? More specifically, all valid styles should not be discarded if some other styles are not found for whatever reason.

Reply:
Yeah, that's a good point, seems the intention of this code is to be robust against both different Format line orderings and presence or absence of fields, but it's currently only defensive against the ordering and breaks if fields we expect to see are absent. This is obviously getting worse as we add support for more style properties.

I don't think this PR is the right place to fix this though - i'll send a separate change. Thanks for flagging!